rpc_jobs/single_use_slave_template.yml

```diff
@@ -22,12 +22,34 @@
               Cleanup
               Destroy Slave
     dsl: |
-      // CIT Slave node
+      library "rpc-gating@${RPC_GATING_BRANCH}"
       common.shared_slave(){
         pubcloud.runonpubcloud {
-          sh """
-            hostname; date
-            echo "your script here"
-          """
-        } //pubcloud
-      } //cit node
+          // Do something that creates an artifact
+          stage("Build"){
+            sh """
+              date > datestamp
+            """
+          }
+
+          stage("Upload"){
+            // upload the artifact to cloudfiles
+            pubcloud.uploadToCloudFiles(
+              container: "jenkins_logs",
+              src: "${WORKSPACE}/datestamp",
+              html_report_dest: "${WORKSPACE}/artifacts.html"
+            )
+          }
+
+          stage("Publish"){
+            // link to artifacts from Jenkins UI
+            publishHTML(
+              alwaysLinkToLastBuild: true,
+              keepAll: true,
+              reportFiles: "artifacts.html",
+              reportName: "Build Artifact Links",
+              reportDir: "${WORKSPACE}"
+            )
+          }
+        }
+      }
```

Comment:
Workspaces are cleaned by the [use_node function](https://github.com/rcbops/rpc-gating/blob/master/pipeline_steps/common.groovy#L591), artifacts will expired out of public cloud in [30 days](https://github.com/rcbops/rpc-gating/blob/master/playbooks/upload_to_cloud_files.yml#L17). [runonpubcloud](https://github.com/rcbops/rpc-gating/blob/master/pipeline_steps/pubcloud.groovy#L118) Handles instance clean up.